src/uct/sm/knem/knem_iface.c

```diff
@@ -60,9 +62,15 @@ static UCS_CLASS_INIT_FUNC(uct_knem_iface_t, uct_md_h md, uct_worker_h worker,
                            const uct_iface_params_t *params,
                            const uct_iface_config_t *tl_config)
 {
+    size_t sys_max_iov;
+
     UCS_CLASS_CALL_SUPER_INIT(uct_base_iface_t, &uct_knem_iface_ops, md, worker,
                               tl_config UCS_STATS_ARG(NULL));
     self->knem_md = (uct_knem_md_t *)md;
+
+    sys_max_iov = ucs_max(sysconf(_SC_IOV_MAX), 1);
```

Comment:
I'm not sure about this because SM/CMA and SM/KNEM might use different limitations for the max_iov. I'm sure it is _SC_IOV_MAX for SM/CMA but didn't find any limitations for knem. 
@shamisp Could you recommend max_iov limitation for knem?